File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;
        HashMap<Integer, Integer> map = new HashMap<>();

        for(int i=0;i<n;i++){
            int comp = target - nums[i];
            if (map.containsKey(comp)) {
                return new int[] { i, map.get(comp) };
            }
            map.put(nums[i], i);
        }
        return new int[] {};
    }
}

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashmap = {}

        for i, num in enumerate(nums):
            comp = target - num

            if comp in hashmap:
                return [hashmap[comp],i]
            hashmap[num] = i
        
        return []
